enterprise/models.py

```diff
@@ -2,42 +2,235 @@
 from django.utils.encoding import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
 
 from model_utils.models import TimeStampedModel
 
 
+class EnterpriseCustomerManager(models.Manager):
+    """
+    Model manager for Enterprise Customer model.
+
+    Filters out inactive Enterprise Customers, otherwise works the same as default model manager.
+    """
+
+    # This manager filters out some records, hence according to the Django docs it must not be used
+    # for related field access. Although False is default value, it still makes sense to set it explicitly
+    # https://docs.djangoproject.com/en/1.10/topics/db/managers/#base-managers
+    use_for_related_fields = False
+
+    def get_queryset(self):
+        """
+        Return a new QuerySet object. Filters out inactive Enterprise Customers.
+        """
+        return super(EnterpriseCustomerManager, self).get_queryset().filter(active=True)
+
+
 @python_2_unicode_compatible
 class EnterpriseCustomer(TimeStampedModel):
     """
-    TODO: replace with a brief description of the model.
+    Enterprise Customer is an organization or a group of people that "consumes" courses.
+
+    Members of Enterprise Customer take courses on the edX platform.
+
+    Enterprise Customer might be providing certain benefits to their members, like discounts to paid course
+    enrollments, and also might request (or require) sharing learner results with them.
+
+    Fields:
+        [implicit] id (IntegerField): EnterpriseCustomer identifier, PRIMARY KEY
+        name (CharField): Enterprise Customer name
+        short_code (AutoSlugField): Enterprise Customer code - used to reference this Enterprise Customer in other
+                                    parts of the system (SSO, ecommerce, analytics etc.)
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
+        members (ManyToMany[User]): list of Enterprise Customer members.
     """
 
-    # TODO: add field definitions
+    NO_DATA_SHARING = "no_sharing"
+    REQUEST_DATA_SHARING = "request"
+    REQUIRE_DATA_SHARING = "require"
+
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomer model.
+        """
+
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer"
+        verbose_name_plural = "Enterprise Customers"
+
+    objects = EnterpriseCustomerManager()
+    all_objects = models.Manager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_("Enterprise Customer name."))
+    short_code = AutoSlugField(
```

Comment:
@clintonb it is unique, non-editable after a record is created and have a sensible default so that operator don't have to manually enter it.

As an added benefit, it is more human-friendly (you can easily go from "ministry-of-defence" to Ministry of Defence record, but not quite easily from "05aa1b9d-33f6-4e92-a3c5-a6c2d99f0999" to whatever else) and allows operator to override it if required.